crowns.lic

```diff
@@ -2,84 +2,128 @@
+
+class Dice
+  include DRC
+  include DRCI
+  include DRCM
+  include DRCT
+
+  def initialize
+    arg_definitions = [
+    [
+      { name: 'bet', regex: /\d+/, optional: true, description: 'Optional override for betting amount.' }
+    ]
+    ]
+
+    settings = get_settings
+    #@junk_list = settings.darkbox_junk.map { |x| /#{x}/i }
+    @trash_items = settings.darkbox_junk
+    @dice_bet_amount = settings.dice_bet_amount || args.bet
+    @dice_money_on_hand = settings.dice_money_on_hand
+    @withdraw = settings.dice_withdraw
+
+    @dice = ["anchor", "crown", "heart", "lightning", "ship", "trident"]
+
+    Flags.add('bet', /"Place your bets!  Place your bets!"/)
+    Flags.add('no_more_bets', "The libik yells")
+
+    dice_game
+  end
+
+  def dice_game
+    loop do
+      check_money
+      walk_to_table
+      play_dice
+      check_prize
     end
   end
-end
 
-loop do
-  count = 0
-  dice.each do |id|
-    case DRC.bput("bet #{dice_bet_amount} on #{id}", /You bet/, /You can't bet right now./, /You don't have that many kronars!/)
-    when /You don't have that many kronars!/
-      DRC.message("******* Out of money! ********")
+  def check_money
+    get_money if check_wealth('kronars') < 200 && @withdraw
+  end
+
+  def get_money
+    DRC.message("*** Heading to bank to money! ***")
+    DRCT.walk_to(16_315)
+    case DRC.bput("withdraw #{@dice_money_on_hand} platinum", 'we are not lending money at this time', 'The clerk counts out')
+    when 'we are not lending money at this time'
       exit
-    when /You bet/
-      count += 1
-    when /You can't bet right now./
-      pause 2 until Flags['bet']
-      Flags.reset('bet')
-      case DRC.bput("bet #{dice_bet_amount} on #{id}", /You bet/, /You can't bet right now./)
-      when /You bet/
-        count += 1
-      when /You can't bet right now./
-        pause 2 until Flags['bet']
-        Flags.reset('bet')
-        DRC.bput("bet #{dice_bet_amount} on #{id}", /You bet/, /You can't bet right now./)
+    end
+    fput('balance')
+  end
+
+  def walk_to_table
+    DRC.message("*** Heading to Dice Table! ***")
+    DRCT.walk_to(16306) unless (DRRoom.title.include?("Crown and Anchor, Table"))
+
+    loop do
+      break if (DRRoom.title.include?("Crown and Anchor, Table"))
+      case DRC.bput("go cerulean curtain", "Reflecting", "As you peer")
+      when "Reflecting"
+        break
+      when "As you peer"
+        case DRC.bput("go carmine curtain", "Reflecting", "As you peer")
+        when "Reflecting"
+          break
+        when "As you peer"
+          DRC.message("*** Waiting for table to open! ***")
+          pause 30
+        end
       end
     end
   end
 
-  if count = 6
-    pause 1 until Flags['no_more_bets']
-    Flags.reset('no_more_bets')
+  def play_dice
+    loop do
+      count = 0
+      @dice.each do |id|
+        case DRC.bput("bet #{@dice_bet_amount} on #{id}", /You bet/, /You can't bet right now./, /You don't have that many kronars!/)
```

Comment:
❓ Is it sufficient to `bet 10 on {thing}` and not need to specify the denomination (i.e. `bet 10 platinum on {thing}`?

Reply:
You are betting coppers.